GeoNode/geonode
# [Fixes #8342] Extend the Map API

references: https://github.com/GeoNode/geonode/issues/8342

## Extend the Map API

Adds the following info to the `maps` endpoint (`/api/v2/maps/<id>/?include[]=data`):

```json
"maplayers": [
    {
        "pk": 9999,
        "extra_params": {},
        "styles": [""],
        "current_style": "",
        "dataset": {
            "alternate": "",
            "featureinfo_custom_template": "",
            "title": "",
            "perms": [],
            "pk": 9999,
            "has_time": false,
            "default_style": {
                "pk": 9999,
                "name": "",
                "workspace": "",
                "sld_title": "",
                "sld_url": ""
            },
            "styles": [
                {
                    "pk": 9999,
                    "name": "",
                    "workspace": "",
                    "sld_title": "",
                    "sld_url": ""
                }
            ],
            "ptype": "",
            "link": "",
            "links": [
                {
                    "extension": "",
                    "link_type": "",
                    "name": "",
                    "mime": "",
                    "url": ""
                }
            ]
        }
    }
]
```

## How to test

* Create a map
* Get the info: *GET* `/api/v2/maps/<id>/?include[]=data`
* Verify the presence of the `maplayers` field and its data. 

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:


The following are required only for core and extension modules (they are welcomed, but not required, for contrib modules):

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.**
